server/routes/profile.js

```diff
@@ -19,4 +20,6 @@ router.route("/update").post(protect, updateProfile);
 
 router.route("/profile-photo").post(uploadImage);
 
+router.route("/payment").post(payment);
```

Comment:
Let's separate this into two routes as mentioned - one to get saved card details for user and one to save card details. Let's also make sure we protect these routes.